### PR TITLE
Add comments before the release

### PR DESCRIPTION
**Pull request by fehiepsi**
TODO:
- [x] Add many more comments to model construction and the usage of M5Data class to manipulate aggregate/disaggregate samples.

### CODE REVIEW

**Comment by fehiepsi** [comment]
@fritzo I have tried to add many comments to the example models. If you want to correct or improve some points, please don't hesitate to push to this branch. I think that with your merge, it is good to have the first release. :)

File: model1.py [hunk 1]
```diff
@@ -2,17 +2,18 @@
...
 This script gives an example on how to use Pyro forecast module to backtest and
-make a submission for M5 accuracy competition.
+make a submission for M5 accuracy/uncertainty competition.
 
 Using the top-down approach in [1], we first construct a model to predict
 the aggregated sales across all items. Then we will distribute the aggregated
 prediction to each product based on its total sales during the last 28 days.
 
-The result is barely better than the best benchmark model ESX from the competition.
+The results are barely better than the best benchmark models from both accuracy
```
Comment (martinjankowiak) [inline model1.py]: do you mean "a little bit better"? "barely" tends to have a negative connotation

File: model1.py [hunk 2]
```diff
@@ -40,35 +42,48 @@
...
+        # encode the additive weekly seasonality
         with pyro.plate("day_of_week", 7, dim=-1):
             seasonal = pyro.sample("seasonal", dist.Normal(0, 5))
         seasonal = periodic_repeat(seasonal, duration, dim=-1)
 
+        # make prediction
         prediction = bias + trend + seasonal + regressor
+        # note that the `pyro.contrib.forecast` module requires that the last
+        # dimension of the data is timeseries dimension;
+        # despite that our data is univariate, we need to follow that requirement
```
Comment (martinjankowiak) [inline model1.py]: this comment is confusing?

File: model1.py [hunk 3]
```diff
@@ -40,35 +42,48 @@
...
+        # and make sure that the last dimension is 1.
         prediction = prediction.unsqueeze(-1)
 
-        # now, we will use heavy tail noise
+        # Now, we will use heavy tail noises because the data has some outlier
```
Comment (martinjankowiak) [inline model1.py]: nit: Now, we will use heavy tail noise because the data has some outliers

File: model2.py [hunk 4]
```diff
@@ -100,16 +110,23 @@ def model(self, zero_data, covariates):
...
+        # alternative: GammaPoisson (or NegativeBinomial, ZeroInflatedNegativeBinomial)
         noise_dist = dist.Gamma(concentration, rate)
 
         with store_plate, product_plate:
             self.predict(noise_dist, mean.new_zeros(mean.shape))
 
 
+# By default, the Forecaster class will use the AutoNormal guide by default.
+# It saves time to construct a guide for our model. However, in some situations,
+# constructing a custom guide is helpful (especially for armotized guides depending
```
Comment (martinjankowiak) [inline model2.py]: typo: armotized

File: model3.py [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+        # Note that the method `m5.aggregate_samples` only aggregates non-aggregated
+        # timeseries to higher levels, but we have timeseries at the middle store+dept
+        # level. So we assume the all items in each department at each store have the
+        # same sales, disaggregate the timeseries at store+dept level to the
+        # non-aggregated level. Then aggretates the result to any higher aggregation level.
+        num_items_by_dept = torch.tensor(m5.num_items_by_dept, device="cpu")
+        pred = pred / num_items_by_dept.unsqueeze(-1).unsqueeze(-1)
+        non_agg_pred = pred.repeat_interleave(num_items_by_dept, dim=-3)
+        non_agg_pred = non_agg_pred.reshape(num_samples, -1, duration)
+        # note that store+dept is the forth level from bottom up.
```
Comment (martinjankowiak) [inline model3.py]: typo: fourth

File: model1.py (hunk 1 above)
Comment (fehiepsi) [reply model1.py]: oh, thanks!

File: model2.py (hunk 4 above)
Comment (fehiepsi) [reply model2.py]: whoa, you have good eyes ;) thanks for detailed reviewing!

**Review by fritzo — APPROVED** [review]

**Pull request merged**